### PR TITLE
Cummax

### PR DESCRIPTION
**Pull request by umairjavaid**
Implemented cummax for jax, torch, numpy, paddle, and tensorflow. Wrote compositional functions for tensorflow and paddle because cummax is not implemented in it. I also wrote the cummax for the array and container. I have tested the backend, all the tests pass. Kindly tell me how to proceed further

### CODE REVIEW

**Comment by umairjavaid** [comment]
solves #15258

File: ivy/functional/ivy/statistical.py [hunk 1]
```diff
@@ -1148,12 +1148,128 @@ def cumprod(
...
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    axis: int = 0,
+    reverse: bool = False,
+    dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    Return the cumulative minimum of the elements along a given axis.
```
Comment (AnnaTz) [inline ivy/functional/ivy/statistical.py:1167]: Great PR! Just a typo here and under axis, where you wrote minimum instead of maximum.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by umairjavaid** [comment]
Hi @AnnaTz,

I have made the requested changes to the pull request. Could you please review it again? Thank you!"

**Review by AnnaTz — APPROVED** [review]
Great, merging this right away!

**Comment by umairjavaid** [comment]
@AnnaTz thank you so much!

**Pull request merged**